scripts/spack/specs.json

```diff
@@ -18,11 +18,13 @@
       "clang@14.0.6+devtools+hdf5+mfem+c2c+scr+profiling",
       "intel@2022.1.0+devtools+hdf5+mfem+c2c+profiling" ],
 
-    "__comment__":"# Use amdgpu_target=gfx90a for tioga/rzvernal; and gfx908 for rznevada",
+    "__comment__":"# Use amdgpu_target=gfx942 for rzadams",
+    "__comment__":"# Use amdgpu_target=gfx90a for tioga/rzvernal",
+    "__comment__":"# Use amdgpu_target=gfx908 for rznevada",
     "__comment__":"# -Wno-int-conversion flag needed for building HDF5",
     "toss_4_x86_64_ib_cray":
-    [ "clang@16.0.0~openmp+mfem+c2c+profiling+rocm amdgpu_target=gfx90a ^hip@5.6.0 ^hsa-rocr-dev@5.6.0 ^llvm-amdgpu@5.6.0 ^rocprim@5.6.0 ^raja~openmp+rocm ^umpire~openmp+rocm ^hdf5 cflags=-Wno-int-conversion",
-      "clang@17.0.0~openmp+mfem+c2c+profiling+rocm amdgpu_target=gfx90a ^hip@5.7.1 ^hsa-rocr-dev@5.7.1 ^llvm-amdgpu@5.7.1 ^rocprim@5.7.1 ^raja~openmp+rocm ^umpire~openmp+rocm ^hdf5 cflags=-Wno-int-conversion" ],
+    [ "rocmcc@6.1.2~openmp+mfem+c2c+profiling+rocm amdgpu_target=gfx942 ^hip@6.1.2 ^hsa-rocr-dev@6.1.2 ^llvm-amdgpu@6.1.2 ^rocprim@6.1.2 ^raja~openmp+rocm ^umpire~openmp+rocm ^hdf5 cflags=-Wno-int-conversion",
+      "rocmcc@6.2.1~openmp+mfem+c2c+profiling+rocm amdgpu_target=gfx942 ^hip@6.2.1 ^hsa-rocr-dev@6.2.1 ^llvm-amdgpu@6.2.1 ^rocprim@6.2.1 ^raja~openmp+rocm ^umpire~openmp+rocm ^hdf5 cflags=-Wno-int-conversion" ],
```

Comment:
Does this mean that `specs.yaml` will no longer be generating host-configs for tioga and rzvernal (`gfx90a`)? 
I think we still want that, especially for our CI.

Reply:
> Does this mean that `specs.yaml` will no longer be generating host-configs for tioga and rzvernal (`gfx90a`)? I think we still want that, especially for our CI.

You can still use the `specs.json`, but you have to manually swap to `gfx90a` in the `specs.json` before calling `build_tpls.py`, as `toss_4_x86_64_ib_cray` is recognized by both rzadams and tioga/rzvernal. There was a period of time when this had to be done to support rznevada and tioga.

I'm not sure what is the better way to handle the two-machine case.
We could some add some additional machine checking in the python scripts (a "hostname" or other environment variable parsing?), either at the spack package recipe level or in `llnl_lc_build_tools.py`.
Not sure if either is the level of visibility we want for something that is part of the spack spec.

This Teams discussion is relevant: [link ](https://teams.microsoft.com/l/message/19:262d8ae6c703498f9b68e42be10a5223@thread.tacv2/1729084718621?tenantId=a722dec9-ae4e-4ae3-9d75-fd66e2680a63&groupId=7b90592d-74bf-4853-a876-53329ee8fa36&parentMessageId=1729084718621&teamName=Spack&channelName=Uberenv&createdTime=1729084718621)

Reply:
We could build a single TPL set that supports both compute architectures by setting `CMAKE_HIP_ARCHITECTURES` to a list of valid architectures.

Reply:
> We could build a single TPL set that supports both compute architectures by setting CMAKE_HIP_ARCHITECTURES to a list of valid architectures.

This works, setting  `amdgpu_target=gfx90a, gfx942` will have the one TPL set work for both rzvernal and rzadams.

> Unfortunately, the Cray Fortran compiler cannot handle multiple architectures. Checking with a few app folks,it seems that they no longer do development or support their codes on MI250X machines because of this.

We are currently testing only `amdclang++` with `amdflang`, so I did not run into this issue.
We removed the `crayCC` with `crayftn`  configs to sort out issues on CI with older rocm versions: #1273

I'll take a look at creating a config with the Cray compilers.

Reply:
Thanks for sorting these issues @bmhan12